code/modules/mob/living/carbon/human/human.dm

```diff
@@ -923,7 +923,7 @@
 	return ..()
 
 /mob/living/carbon/human/is_literate()
-	return TRUE
+	return !has_trauma_type(/datum/brain_trauma/mild/illiterate, TRAUMA_RESILIENCE_ABSOLUTE)
```

Comment:
Complex and redefined getters are not great, and ideally we don't expand on them.
Could you implement this as a mob variable or trait, instead of expanding `is_literate()`?
If you take a look at the getters and setters part of the contribution guidelines you'll have an example on how to do it.
If you want more help here, feel free to ping me in the coder channel.

Reply:
Here is why it would be a bad idea to do it as a trait:  Current functionality is, mobs are illiterate, unless they are a human, silicon, or observer.  Using a trait to handle the same would be quite difficult.  If changing most mobs to be literate is an acceptable change to go along with this, then I am willing to do so.  I just did not want to include a potentially massive balance change alongside this new feature.

As far as a mob variable, if we add future causes of illiteracy, such as a round start quirk, then it would require intricate code to prevent curing a trauma from getting rid of the round start quirk.

Reply:
Can you just make LITERATE a trait instead? So all is_literate() calls become HAS_TRAIT(LITERATE)?

Reply:
Adding it as a trait wouldn't require you to make balance changes, you can handle it during a mob's initialization and preserve current behavior.

The variable method would be better, as it would have higher extensibility.
Now, let's follow your logic and say in the future someone adds a roundstart quirk which can make you illiterate, so the human literacy check has to check both for this quirk and for a trauma. And let's say I want to react to the event of literacy changing. I then have no way to do this.

The idea is to make the code more events-based, and that's why I'm suggesting a variable. If a second source of illiteracy is added then you can add the appropriate checks on setter procs and we can easily append behavior as required.

Reply:
> Now, let's follow your logic and say in the future someone adds a roundstart quirk which can make you illiterate, so the human literacy check has to check both for this quirk and for a trauma.

This comment aged well.